drivers/test-fw/scripts/vector_gen/src/preconditioned_key_extract_gen.rs

```diff
@@ -0,0 +1,84 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    preconditioned_key_extract_gen.rs
+
+Abstract:
+
+    A vector generator for the preconditioned_key_extract function.
+
+--*/
+
+use crate::utils::*;
+
+use openssl::symm::{encrypt, Cipher};
+
+pub struct PreconditionedKeyExtractVector {
+    pub input_key: [u8; 48],
+    pub salt: [u8; 48],
+    pub kdf_label: Vec<u8>,
+    pub checksum: [u8; 16],
+    pub kdf_output: [u8; 64],
+    pub output_key: [u8; 64],
+    pub fingerprint: [u8; 16],
```

Comment:
```suggestion
    pub input_key: [u8; 48],
    pub input_salt: [u8; 48],
    pub input_kdf_label: Vec<u8>,
    pub checksum: [u8; 16],     // AES-ECB-Encrypt(input_salt[0..32], 0x0000...)
    pub kdf_output: [u8; 64],   // KDF(input_key, input_kdf_label, checksum)
    pub output_key: [u8; 64],   // HMAC(input_salt, kdf_output)
    pub fingerprint: [u8; 16],  // AES-ECB-Encrypt(output_key[0..32], 0x0000...)
```
This would help readers understand the shape of the vectors.